PlatformConfigurations/IEEE802.1xPortBasedAuthentication.md

```diff
@@ -0,0 +1,872 @@
+---
+title: IEEE 802.1x (Port Based Authentication)
+parent: Platform Configuration
+nav_order: 5
+layout: default
+---
+
+# IEEE 802.1x (Port Based Authentication)
+
+## Introduction
+
+In this guide, we will quickly explain what IEEE 802.1x (Port Based Authentication) is
+and give an example implementing it.
+
+IEEE 802.1x is a standard for port-based network access control that provides an authentication
+mechanism to devices wishing to join a LAN or WLAN. IEEE 802.1x uses the Extensible Authentication Protocol
+(EAP) to transmit authentication information over ethernet frameworks. The three main components
+include the **Supplicant, Authenticator, and Authentication Server**.
+
+- **Supplicant**: The Supplicant is the device attempting to access the network. It is often the one that
+  participates in the EAP transaction with the Authenticator.
+
+- **Authenticator**: The Authenticator is the network device that connects the supplicant to the network. It acts as the
+  access point for the Supplicant and controls the type of traffic flow between the Supplicant and
+  the network before and after authentication. The Authenticator sends
+  EAP requests to the Supplicant to identify a connecting device.
+  In IEEE 802.1x, this is also more broadly refered to as a Network Access Server.
+
+- **Authentication Server**: The Authentication Server receives requests for network access
+  and verifies devices attempting to connect to the network. This information is then passed back to the authenticator
+  where traffic flow is changed based on the results. This is also commonly refered to as a RADIUS Server.
+  It is also often known as an AAA (Authentication, Authorization, and Accounting) Server.
+
+The image below is a good visualization of the 802.1x Progression.
+
```

Comment:
We should explain how the protocol works here. It is worth including the following info
- when a device is connected the first time, only protocol it can use is the EAP, all other communication is blocked until the device is authenticated.
- Dent requests identity of the device asking for access and sends it to the authentication server.
- The authentication server uses the identity information in its user database. If a matching entry is found, 
--if it is a password authentication, the server sends a challenge to the access device, and the device will respond with the encrypted password (you don't need to go into the details). The authentication server compares the received password what it has for that user identity. If they are the same the server considers the client valid and sends a RADIUS Access-Accept packet to the access device.
--if it is a MAC authentication, ....

Reply:
Also add info on where all the SW you are installing/setting up (freeradius, hostapd,  wpa_supplicant, iproute2) are going to be at. It is better if you accompany these descriptions with a graphical representation.

Reply:
> We should explain how the protocol works here. It is worth including the following info
> 
> * when a device is connected the first time, only protocol it can use is the EAP, all other communication is blocked until the device is authenticated.
> * Dent requests identity of the device asking for access and sends it to the authentication server.
> * The authentication server uses the identity information in its user database. If a matching entry is found,
>   --if it is a password authentication, the server sends a challenge to the access device, and the device will respond with the encrypted password (you don't need to go into the details). The authentication server compares the received password what it has for that user identity. If they are the same the server considers the client valid and sends a RADIUS Access-Accept packet to the access device.
>   --if it is a MAC authentication, ....

The latest commit will now add a section describing the protocol in the introduction:
![image](https://github.com/user-attachments/assets/c2163700-840c-4c5f-aa80-5e2a93a3fbe4)

Reply:
> Also add info on where all the SW you are installing/setting up (freeradius, hostapd, wpa_supplicant, iproute2) are going to be at. It is better if you accompany these descriptions with a graphical representation.

I have reworded the Installation section to reference the image of the example configuration to help users understand on what device each utility needs to be installed on:

![image](https://github.com/user-attachments/assets/80c25543-d508-44a1-be12-7d656e0cc655)